mlir/lib/Conversion/GPUCommon/GPUOpsLowering.h

```diff
@@ -35,16 +35,42 @@ struct GPUDynamicSharedMemoryOpLowering
   unsigned alignmentBit;
 };
 
+struct GPUFuncOpLoweringOptions {
+  /// The address space to use for `alloca`s in private memory.
+  unsigned allocaAddrSpace;
+  /// The address space to use declaring workgroup memory.
+  unsigned workgroupAddrSpace;
```

Comment:
Just a thought: why is it that we only propagate specific ones and not an entire address space map? The lowering would need information on other types as well anyway.

Reply:
This is just for the `gpu.func` attributions. Regular ones should be mapped using attribute conversion.